File: pesocial-frontend/src/services/authService.ts
import api from './api'
import type { AuthResponse, LoginRequest, RefreshTokenRequest, RegisterRequest } from '../types/auth'

export const authService = {
  login: async (payload: LoginRequest) => {
    const { data } = await api.post<AuthResponse>('/api/auth/login', payload)
    return data
  },

  register: async (payload: RegisterRequest) => {
    const { data } = await api.post<AuthResponse>('/api/auth/register', payload)
    return data
  },

  refresh: async (payload: RefreshTokenRequest) => {
    const { data } = await api.post<AuthResponse>('/api/auth/refresh', payload)
    return data
  },

  logout: async (refreshToken: string) => {
    await api.post('/api/auth/logout-service', { refreshToken })
  },
}

File: src/main/java/com/pesocial/service/AuthService.java
package com.pesocial.service;

import com.pesocial.dto.auth.AuthResponse;
import com.pesocial.dto.auth.LoginRequest;
import com.pesocial.dto.auth.LogoutRequest;
import com.pesocial.dto.auth.RefreshTokenRequest;
import com.pesocial.dto.auth.RegisterRequest;

public interface AuthService {
    AuthResponse register(RegisterRequest request);
    AuthResponse login(LoginRequest request);
    AuthResponse refresh(RefreshTokenRequest request);
    void logout(LogoutRequest request);
}
